### PR TITLE
BUG-7023 allow error bars to use style

### PR DESCRIPTION
**Pull request by GLeurquin**
- [x] closes #7023
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -728,7 +728,7 @@ Plotting
 
 - Bug in :func:`scatter_matrix` raising when 2d ``ax`` argument passed (:issue:`16253`)
 - Prevent warnings when matplotlib's ``constrained_layout`` is enabled (:issue:`25261`)
--
+- Bug in :meth:`MPLPlot._plot` ignoring ``style`` when using error bars (:issue:`7023`)
```
Comment (mzeitlin11) [inline doc/source/whatsnew/v1.3.0.rst:731]: Can you describe this in a user-facing way (what changes about publicly exposed pandas api?)

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @GLeurquin. One comment on whatsnew, plus needs tests!

**Review by jreback — CHANGES_REQUESTED** [review]
pls also adds a test that exercises this code path

**Comment by datapythonista** [comment]
Closing as stale. @GLeurquin if you want to add tests and address the comment, just ping me, and more than happy to reopen.

**Pull request closed without merging**